config/registry_image_pruner/image_pruner/prune_images.py

```diff
@@ -47,11 +47,11 @@ def delete_image_repo(quay_token: str, namespace: str, name: str, tag: str) -> N
 
 def remove_images(images: Dict[str, Any], quay_token: str, namespace: str, name: str, dry_run: bool = False) -> None:
     image_digests = [image["manifest_digest"] for image in images.values()]
+    image_regex = re.compile(r"^sha256-([0-9a-f]+)(\.sbom|\.att)$", re.IGNORECASE)
     for image in images:
         # attestation or sbom image
-        if image.endswith(".att") or image.endswith(".sbom"):
-            sha = re.search("sha256-(.*)(.sbom|.att)", image).group(1)
-            if f"sha256:{sha}" not in image_digests:
+        if (match := image_regex.match(image)) is not None:
```

Comment:
Is `image` just a tag? If it's a full image name, it won't match something that starts with `sha256-...`

Reply:
ran the script on my `susdas_quay_org` quay repo, the script is not removing `.sbom` or `.att` tags, as it needs to be, below is an example images dictionary which is passed to the `remove_images` method 
```
images = {'sha256-876b86c27ac4132a1660f7325fd2350a6b9f792ce30a87cfa626753b7ca98a3d.att': {'name': 'sha256-876b86c27ac4132a1660f7325fd2350a6b9f792ce30a87cfa626753b7ca98a3d.att', 'size': 89162, 'last_modified': 'Mon, 21 Aug 2023 07:09:56 -0000', 'manifest_digest': 'sha256:52508a6193fdfa67f125f47a632ee6420f34d729776752890a5c6e00dbad6725'}, 
          'sha256-876b86c27ac4132a1660f7325fd2350a6b9f792ce30a87cfa626753b7ca98a3d.sig': {'name': 'sha256-876b86c27ac4132a1660f7325fd2350a6b9f792ce30a87cfa626753b7ca98a3d.sig', 'size': 307, 'last_modified': 'Mon, 21 Aug 2023 07:08:33 -0000', 'manifest_digest': 'sha256:0e3c6b8cdde71673336dc988425dbc0283445ae72878a675300399b31599e525'}, 
          'sha256-876b86c27ac4132a1660f7325fd2350a6b9f792ce30a87cfa626753b7ca98a3d.sbom': {'name': 'sha256-876b86c27ac4132a1660f7325fd2350a6b9f792ce30a87cfa626753b7ca98a3d.sbom', 'size': 114547, 'last_modified': 'Mon, 21 Aug 2023 07:08:29 -0000', 'manifest_digest': 'sha256:7cd13d1b83d45e7f7a9c526b2264c224c858f03f3dd27eaa4980fc44a0c50709'}, 
          'build-8ca40-1692601660': {'name': 'build-8ca40-1692601660', 'size': 16091103, 'last_modified': 'Mon, 21 Aug 2023 07:08:27 -0000', 'manifest_digest': 'sha256:876b86c27ac4132a1660f7325fd2350a6b9f792ce30a87cfa626753b7ca98a3d'}
}
```

Reply:
> `image` is tag.

Could you maybe rename the variable to make it clearer?

Reply:
@tkdchen then does the `prune_images.py` script doing what it is supposed to (cleaning up of .att and .sbom image tags) ?

Reply:
I would not say that `image` is a tag. It's separate image even though it is under the "tags" section in quay (this is just the rhtap/quay integration design). The .att .sbom images are associated with the build image, but they are not  just tags for that image and they should be removed

Reply:
It would be nice to have a unit test specifically for remove images, where various scenarios would be tested like the old unit tests had so we can verify it works as expected

Reply:
I can now confirm that the script does what it is supposed to
```
(venv) [mkosiarc@fedora image_pruner]$ git branch
  image-pruner
  main
* tkdchen-STONEBLD-1785-pruner
# nothing to do, build image present
(venv) [mkosiarc@fedora image_pruner]$ QUAY_TOKEN=<TOKEN> python prune_images.py --namespace mkosiarc_rhtap
2024-01-09 14:57:33,957 - INFO - Processing repository 0: mkosiarc_rhtap/test-pvc/test-application/sample-app-newer-python-image
# I removed the build image, the script should remove the .att and .sbom images
(venv) [mkosiarc@fedora image_pruner]$ QUAY_TOKEN=<TOKEN> python prune_images.py --namespace mkosiarc_rhtap
2024-01-09 14:57:49,377 - INFO - Processing repository 0: mkosiarc_rhtap/test-pvc/test-application/sample-app-newer-python-image
2024-01-09 14:57:50,224 - INFO - Removing image sha256-f685cce54b2d4491a4a8b7a3bc6db437190a28795684ebe8271c8d3ad6cf9120.att from mkosiarc_rhtap/test-pvc/test-application/sample-app-newer-python-image
2024-01-09 14:57:51,308 - INFO - Removing image sha256-f685cce54b2d4491a4a8b7a3bc6db437190a28795684ebe8271c8d3ad6cf9120.sbom from mkosiarc_rhtap/test-pvc/test-application/sample-app-newer-python-image
```

Reply:
Hi @mkosiarc, thanks for sharing your thought. Regarding the unit tests, `remove_images` has been covered 100% already.

Reply:
> > `image` is tag.
> 
> Could you maybe rename the variable to make it clearer?

Renamed.

Reply:
Thanks chen for adding those tests, I appreciate it.